code/__HELPERS/game.dm

```diff
@@ -379,3 +379,39 @@
 	var/dest_y = src_y + distance*cos(rotation);
 
 	return new /datum/projectile_data(src_x, src_y, time, distance, power_x, power_y, dest_x, dest_y)
+
+/proc/getRotationFromDirs(dir1, dir2)
+	return getDegreesFromDir(dir2)-getDegreesFromDir(dir1)
+
+/proc/getDirFromRotation(rotation, olddir)
+	return getDirFromDegrees(rotation+getDegreesFromDir(olddir))
+
+/proc/getDirFromDegrees(degrees)
+	if (degrees < 0)
+		degrees += 360
+	if (degrees >= 360)
+		degrees %= 360
+	if (degrees == 360)
+		degrees = 0
+	switch(degrees)
+		if (0)
+			return NORTH
+		if (90)
+			return EAST
+		if (180)
+			return SOUTH
+		if (270)
+			return WEST
+
+
+/proc/getDegreesFromDir(dir)

```

Comment:
it could happen, but anywho, I did what you asked and have it set to round up if the rotation angle isn't a mod of 90.